### DISCUSSION TITLE
Syntax highlighting of PHP heredoc strings

### DISCUSSION THREAD

**Discussion by sylbru** [zed-industries/zed#18239] [Feature Requests]
Hi! Starting a discussion here about a tiny but nice change I want to propose on the php extension.

PHP "heredoc" strings make it easy to define string literals over multiple lines:

```php
    $someString = <<<EOT
        multiline
        text
        EOT;
```

That `EOT` identifier can be anything else, and it is actually being used in Sublime Text and VS Code to inject syntax highlighting for another language in said string, depending on the identifier. For instance, if the identifier is SQL, SQL syntax highlighting will be applied to the contents of the string. Likewise if the identifier is CSS or JS.

```php
    $someString = <<<SQL
        SELECT *
        FROM my_table
        SQL;
```

The change is super simple, it consists of one line added to `injections.scm`:

```
((heredoc_body) (heredoc_end) @language) @content
```

Should I wait for feedback or create my PR right away? As it seems very straightforward to me, and not requiring much debate.

**Comment by notpeter** [comment]
Always happy to review PRs, especially small ones.

Not sure the implications (e.g. does it error if you use something that's not a language?) but nifty none the less.

**Reply by sylbru** [reply]
Thanks, I just did that! (with mostly the same text as here) https://github.com/zed-industries/zed/pull/18368

**Reply by sylbru** [reply]
I haven’t seen any errors even in the logs when using an identifier that doesn’t match any existing language.

**Comment by nertzy** [comment]
This would also be great for shell script heredocs, e.g.:

```shell
psql -d postgres <<SQL
SELECT *
FROM my_table;
SQL
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
